Doc/whatsnew/3.9.rst

```diff
@@ -115,6 +115,49 @@ to easily remove an unneeded prefix or a suffix from a string. Corresponding
 added. See :pep:`616` for a full description. (Contributed by Dennis Sweeney in
 :issue:`18939`.)
 
+PEP 585: Builtin Generic Types
+------------------------------
+
+In type annotations you can now use built-in collection types such as
+``list`` and ``dict`` as generic types instead of importing the
+corresponding capitalized types (e.g. ``List`` or ``Dict``) from
+``typing``.  Some other types in the standard library are also now generic,
+for example ``queue.Queue``.
+
+Example:
+
+.. code-block:: python
+
+   def greet_all(names: list[str]) -> None:
+       for name in names:
+           print("Hello", name)
+
+See :pep:`585` for more details.  (Contributed by Guido van Rossum,
+Ethan Smith, and Batuhan Taşkaya in :issue:`39481`.)
+
+PEP 617: New Parser
+-------------------
+
+Python 3.9 uses a new parser, based on `PEG
+<https://en.wikipedia.org/wiki/Parsing_expression_grammar>`_ instead
+of `LL(1) <https://en.wikipedia.org/wiki/LL_parser>`_.  The new
+parser's performance is roughly comparable to that of the old parser,
+but the PEG formalism is more flexible than LL(1) when it comes to
+designing new language features.  We'll start using this flexibility
+in Python 3.10 and later.
```

Comment:
Are there no new features that PEG allowed in the current state? Is this 100% 1:1 with the LL-1 parser?

One thing you might mention is why this work was done in the first place. "It's more flexible" isn't the entire truth. The truth is that Python's actual grammar even pre-3.9 was not really LL-1 anymore and the current parser overstayed its welcome, leading to tricky maintenance.

Reply:
There are two Easter eggs; the new parser produces a syntax error if you use `__new_parser__` (this is for us so we can ensure a code path is actually using the new parser), and it supports `with (ctxmgr1 as var1, ctxmgr2 as var2, ...): block`. But I think neither should be documented (and maybe we should remove the latter during the beta cycle, before people start relying on it).

I'd prefer not to go into a long discussion about the "why" here, people can read the PEP. Honestly, it's hardly worth a mention in "what's new" except people need to be aware and we need to advertise the ways to disable it.

Reply:
> maybe we should remove the latter during the beta cycle, before people start relying on it

Please no, that was one of the motivators behind me nagging you about this at the core sprint in 2017 and 2018 :-)